### ISSUE TITLE
Schema View: 2 views with the same interface override each other

### ISSUE THREAD

**Issue by islamaliev** [sourcenetwork/defradb#2182]
When creating 2 view that have interfaces with the same name, then one overrides another and the overridden one is not available in the first view anymore.
```go
func TestView_Override(t *testing.T) {
	test := testUtils.TestCase{
		Description: "One to many view with alias on outer object",
		Actions: []any{
			testUtils.SchemaUpdate{
				Schema: `
					type Author {
						name: String
						books: [Book]
					}
					type Book {
						name: String
						author: Author
					}
				`,
			},
			testUtils.CreateView{
				Query: `
					Author {
						name
						books {
							name
						}
					}
				`,
				SDL: `
					type AuthorView {
						name: String
						books: [BookView]
					}
					interface BookView {
						name: String
					}
				`,
			},
			testUtils.CreateView{
				Query: `
					Author {
						authorName: name
						books {
							bookName: name
						}
					}
				`,
				SDL: `
					type AuthorAliasView {
						authorName: String
						books: [BookView]
					}
					interface BookView {
						bookName: String
					}
				`,
			},
			// bae-ef9cd756-08e1-5f23-abeb-7b3e6351a68d
			testUtils.CreateDoc{
				CollectionID: 0,
				Doc: `{
					"name":	"Harper Lee"
				}`,
			},
			testUtils.CreateDoc{
				CollectionID: 1,
				Doc: `{
					"name":	"To Kill a Mockingbird",
					"author_id": "bae-ef9cd756-08e1-5f23-abeb-7b3e6351a68d"
				}`,
			},
			testUtils.Request{
				Request: `query {
							AuthorAliasView {
								authorName
								books {
									bookName
								}
							}
						}`,
				Results: []map[string]any{
					{
						"authorName": "Harper Lee",
						"books": []map[string]any{
							{
								"bookName": "To Kill a Mockingbird",
							},
						},
					},
				},
			},
			// This action results in error: Cannot query field "name" on type "BookView".
			/*testUtils.Request{
				Request: `query {
							AuthorView {
								name
								books {
									name
								}
							}
						}`,
				Results: []map[string]any{
					{
						"name": "Harper Lee",
						"books": []map[string]any{
							{
								"name": "To Kill a Mockingbird",
							},
						},
					},
				},
			},*/
			// This action returns books with "name" field although "bookName" is requested
			testUtils.Request{
				Request: `query {
							AuthorView {
								name
								books {
									bookName
								}
							}
						}`,
				Results: []map[string]any{
					{
						"name": "Harper Lee",
						"books": []map[string]any{
							{
								"bookName": "To Kill a Mockingbird",
							},
						},
					},
				},
			},
		},
	}

	testUtils.ExecuteTestCase(t, test)
}
```
This issue in this example is minor in my opinion but it seems like there might be more severe implications.

EDIT: After this creation of other views is broken:
```go
			// This action returns error: no type found for given name. Type: BookView.
			testUtils.CreateView{
				Query: `
					Author {
						name
					}
				`,
				SDL: `
					type AuthorDummyView {
						name: String
					}
				`,
			},
```

**Comment by AndrewSisley** [comment]
Nice catch Islam! Will look into this one shortly, I'm also worried that adding an interface with a name of an existing collection might result in similar issues as it looks like the schema is not checking for existing schema of that name before writing to store.

**Comment by AndrewSisley** [comment]
> After this creation of other views is broken

This is strange, it might be there is a weakness in the descriptions-gql code that permits an in-memory corruption of state.  Might be worth looking into how this works later.

**Comment by AndrewSisley** [comment]
> After this creation of other views is broken

This is a separate issue, unrelated to the overwriting problem.  I've opened up https://github.com/sourcenetwork/defradb/issues/2184

**Comment by fredcarle** [comment]
closed by #2185 2185
